### ISSUE TITLE
0.13.0 build 1 is broken on linux

### ISSUE THREAD

**Issue by grlee77** [conda-forge/pyfftw-feedstock#51]
Bug report at https://github.com/pyFFTW/pyFFTW/issues/294#issuecomment-1024595535

I can recreate the issue locally. If I force install of build 0 which was not using the static linking, then it works okay for me. 

Specifically the following version works, but build "1" does not
`mamba install "pyfftw=0.13.0=py39h51d1ae8_0"`  

The PyPI packages also work as expected.

**Comment by grlee77** [comment]
not sure if the same issue is also present on OS X

**Comment by grlee77** [comment]
Build 1 was where we switched back to static linking, so apparently something went wrong there. We can expand the test case a bit and see if the issue is reproduced on CI.

**Comment by paugier** [comment]
I also experienced this bug. It would be nice to find a solution :slightly_smiling_face: because this bug should affect a lot of packages and users for "simple installations".

For example on Linux, `conda create -n env_pyfftw pyfftw` leads to a broken environment.

**Comment by jakirkham** [comment]
Should we just revert the static linking?

**Comment by grlee77** [comment]
That would be easiest. I didn't try looking at the logs to see what could be going wrong. We should also add a minimal test case like in the linked issue.

**Comment by grlee77** [comment]
The error is missing symbols that seem related to `pthreads` e.g. [from this run](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=462268&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642)
```
  threads.c:(.text.fftw_threads_register_planner_hooks+0x40): undefined reference to `sem_init'
  collect2: error: ld returned 1 exit status
```
that results in a single-threaded build being run. There is perhaps an upstream bug in this single-threaded case, but that is not what it should be building. We have set the recipe to use `pthreads` multithreading (as opposed to OpenMP). I tried manually adding `-pthread` to CFLAGS in the build script, but it did not help (this should not be necessary as `setup.py` is already supposed to add it).

Also, there are `RuntimeError`s during `run_tests.py` but this surprisingly does not cause the build to fail?

When I remove the `STATIC_FFTW_DIR` argument, the build succeeds again and no errors are run during testing. The log confirms that pthreads version is being built:
```
2022-02-20T01:49:10.6692980Z   INFO:__main__:Build pyFFTW with support for FFTW with
2022-02-20T01:49:10.6694033Z   INFO:__main__:double precision + pthreads
2022-02-20T01:49:10.6695354Z   INFO:__main__:single precision + pthreads
2022-02-20T01:49:10.6696310Z   INFO:__main__:long precision + pthreads
```

**Comment by grlee77** [comment]
This makes me wonder if there is an issue with the static FFTW libs from `fftw-feedstock`

**Comment by gdonval** [comment]
I don't know if it is related but I am still seeing the original message about undefined plan with nthreads.

**Comment by paugier** [comment]
> Should we just revert the static linking?

Is there a reason not to do so? build 0 seems to work fine.

**Comment by al-zatv** [comment]
> I don't know if it is related but I am still seeing the original message about undefined plan with nthreads.

Same here.

**Comment by josh-willis** [comment]
> > Should we just revert the static linking?
> 
> Is there a reason not to do so? build 0 seems to work fine.

Will reverting the static linking on linux not cause the bug fixed in #5 to resurface, if numpy is linked against MKL and numpy is imported first?

**Comment by paugier** [comment]
Currently the last build with static linking is not usable on Linux. Coming back to dynamic linking might lead to a bug for numpy linked with MKL but conda-forge users won't see that by default. It seems to me that it is better to have something working for the simple case that something not working at all.
